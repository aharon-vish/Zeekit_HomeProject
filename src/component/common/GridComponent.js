import React from 'react';
import _ from 'lodash';

const Grid = (props) => {
    const {items, styles, onMouseEnter, onMouseLeave, onGridItemClick} = props;

    return ( <div style={styles ? styles.gridStyle : null}>
        {renderItems(items, styles, onMouseEnter, onMouseLeave, onGridItemClick)}
    </div>)
};

const renderItems = (items, styles, onMouseEnter, onMouseLeave, onGridItemClick) => {
    return _.map(items, function (item, index) {
        if (item.Title && item.Year && item.Poster) {
            return (
                <div className={styles.gridItem} key={item.imdbID}>
                <h5>{item.Title}</h5>
                <h6>{item.Year}</h6>
                <img alt="Image Not found" title="Image Not found Images" src={item.Poster}/>
            </div>)

        } else {
            return (<div
                onClick={onGridItemClick}
                onMouseEnter={onMouseEnter}
                onMouseLeave={onMouseLeave}
                style={styles ? styles.gridItem : null}
                key={index}>{item}</div>)
        }
    });
};
export default Grid;