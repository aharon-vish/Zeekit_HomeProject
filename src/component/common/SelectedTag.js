import React from 'react';
import _ from 'lodash';
const SelectedTag = (props) => {

    const renderOption = (filter)=> {
        return _.map(filter, function (itemFilter) {
            return (<option key={itemFilter} value={itemFilter}>{itemFilter}</option>)
        });
    };

    return (
        <div className="optionalTags">
            <div>Optional Tags:</div>
            <ul>
                <li>Year:&nbsp;
                    <select onChange={props.onFilterChange} data-filter-type="year">
                        {renderOption(props.filterYear)}
                    </select>
                </li>
                <li>Type:&nbsp;
                    <select onChange={props.onFilterChange} data-filter-type="type">
                        {renderOption(props.filterType)}
                    </select>
                </li>
            </ul>
        </div>
    )
};

export default SelectedTag;