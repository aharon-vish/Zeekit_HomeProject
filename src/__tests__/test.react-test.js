import React from 'react';
import ReactDOM from 'react-dom';
import Title from '../component/common/TitleComponent';
import GridComponent from '../component/common/GridComponent';
import SelectedTag from '../component/common/SelectedTag';
import renderer from 'react-test-renderer';

describe('Title', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Title/>, div);
    });
});
describe('GridComponent', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<GridComponent/>, div);
    });
});
test('Title changes the class when hovered', () => {
    const component = renderer.create(
        <Title text="test"></Title>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('Title changes the class when hovered', () => {
    const component = renderer.create(
        <SelectedTag/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onChange();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});