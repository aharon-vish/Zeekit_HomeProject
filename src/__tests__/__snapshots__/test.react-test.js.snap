// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title changes the class when hovered 1`] = `
<h1
  style={undefined}
>
  test
</h1>
`;

exports[`Title changes the class when hovered 2`] = `
<div
  className="optionalTags"
>
  <div>
    Optional Tags:
  </div>
  <ul>
    <li>
      Year: 
      <select
        data-filter-type="year"
        onChange={undefined}
      />
    </li>
    <li>
      Type: 
      <select
        data-filter-type="type"
        onChange={undefined}
      />
    </li>
  </ul>
</div>
`;
